app/src/pages/contract/[chain]/[address].tsx

```diff
@@ -128,18 +82,39 @@ export default function Address() {
   const [isLoadingAddressInfos, setIsLoadingAddressInfos] = useState(false)
   const [addressInfosSearch, setAddressInfosSearch] = useState('')
   const [isAttestationInfosOpen, setIsAttestationInfosOpen] = useState(false)
+  const [isLoadingAttestations, setIsLoadingAttestations] = useState(false)
+  const [attestationsKnownUsers, setAttestationsKnownUsers] = useState<
+    ContractAttestation[]
+  >([])
+  const [attestationsStrangers, setAttestationsStrangers] = useState<
+    ContractAttestation[]
+  >([])
+
+  const [attesting, setAttesting] = useState(false)
+  const [isStale, setIsStale] = useState(false)
+  const [signer, setSigner] = useState<ethers.Signer | null>(null)
+
+  const contractHash = useMemo(() => {
+    if (!contracts.length) return ''
+    return objecthash(contracts[0].abi)
```

Comment:
This should contain source code of all contracts that belong to this address
```suggestion
    return objecthash(contracts.map(c => c.sourceCode))
```